modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java

```diff
@@ -1164,8 +1178,14 @@ public void paint(final Graphics2D graphics, final GridCoverage2DReader gridCove
 
         setupInterpolationHints(interpolation);
         
+        //
+        // Band selection
+        //
+        applyBandsSelectionParameterIfNeeded(gridCoverageReader, readParams, symbolizer);
+        RasterSymbolizer finalSymbolizer = setupSymbolizerIfBandsSelectionIsUsed(symbolizer);
```

Comment:
Aren't you modifying the symbolizer also if the bands parameters are not supported in the reader this way? Please add a test that renders a coverage from a reader with channel selection (e..g, select the second band out of a RGB sources) and make sure this code is not interfering (e.g, you are getting back the green one, not the red one, suggestion, create a compressed geotiff that's fully green to simplify the test checks)